Software Documentation/Naming and Coding Style Conventions.md

```diff
@@ -0,0 +1,48 @@
+### Variables
+- Constants
+    - Preceded by `k`
+    - First word should be the name of the subsystem the constant is for
+    - Ex: `kDriveLeftRearTalonId`, `kLimelightFov`, `kControlLoopPeriod`
+- Fields
+    - Preceded by `m`
+    - Fields are any variables found in the **class body**
+    - Ex: `mWristAngle`, `mGyroAngle`
+- Parameters
+    - Preceded by `p`
+    - Ex:
+    ```
+        public DriveStraight(Drive pDrive, double pDistanceToDrive) {
+            mDrive = pDrive;
+            mDistanceToDrive = pDistanceToDrive;
+        }
+    ```
+
+### Constructors
+- If your class needs to use another classes' object, it should be passed into the constructor
+- Settings that are essential to your code's functionality (like the number of degrees to turn for `GyroTurn`) can also be passed in
+
+### Setting Properties
+- If you need to override a default setting for a class, use the flow model
```

Comment:
What do you mean by the flow model? I can infer from what it's doing, but I'm not sure this is covered for new developers. Focus on the pattern itself (flow pattern is when the setter method returns this.)